### PR TITLE
Exercise: Write chunk(-like data structure) to disk and read it using mmap.

### PR DESCRIPTION
**Pull request by phoeinx**
Adapted according to feedback from @tjjordan and me.

### CODE REVIEW

File: src/bin/playground.cpp [hunk 1]
```diff
@@ -1,10 +1,129 @@
...
+#include <numeric>
+#include <span>
+#include <vector>
 #include "types.hpp"
 
+#include <unistd.h>
+
 using namespace hyrise;  // NOLINT
 
+#define chunk_prototype std::vector<std::shared_ptr<std::vector<uint32_t>>>
```
Comment (Bouncner) [inline src/bin/playground.cpp:14]: Prefer `using` over #define.

**Review by Bouncner — COMMENTED** [review]
Awesome.

File: src/bin/playground.cpp [hunk 2]
```diff
@@ -97,12 +98,15 @@ int main() {
   const auto COLUMN_COUNT = uint32_t{23};
   const auto ROW_COUNT = uint32_t{65000};
```
Comment (Bouncner) [inline src/bin/playground.cpp:99]: Easier to read: `65'000`

Comment (Bouncner) [reply src/bin/playground.cpp:99]: Btw. In Hyrise, we use upper case names for constexpr variables.

File: src/bin/playground.cpp [hunk 3]
```diff
@@ -45,40 +45,41 @@ chunk_prototype create_chunk(const uint32_t row_count, const uint32_t column_cou
...
+  Assert((map != MAP_FAILED), fail_and_close_file(fd, "Mapping Failed: ", errno));
+  close(fd);
 
-    madvise(map, chunk_bytes, MADV_SEQUENTIAL);
+  madvise(map, chunk_bytes, MADV_SEQUENTIAL);
 
+  for (auto column_index = size_t{0}; column_index < column_count; ++column_index) {
     // create std::span view on map
-    auto map_span_view = std::span{map, chunk_size};
+    auto map_span_view = std::span{map + (column_index * segment_size), segment_size};
```
Comment (Bouncner) [inline src/bin/playground.cpp:82]: That was fast. Nice!

**Comment by rrcomtech** [comment]
Merged master to have a working CI config.

**Comment by phoeinx** [comment]
Closing for now to clean up PR board.

**Pull request closed without merging**